k4MarlinWrapper/examples/event_display.py

```diff
@@ -18,58 +18,53 @@
 # limitations under the License.
 #
 
-from Gaudi.Configuration import *
+import sys
+
+from Gaudi.Configuration import INFO
+from Configurables import MarlinProcessorWrapper, k4DataSvc, PodioInput, EDM4hep2LcioTool, GeoSvc
+from k4FWCore.parseArgs import parser
+from k4MarlinWrapper.inputReader import create_reader, attach_edm4hep2lcio_conversion
+
+
+parser.add_argument(
+    "--inputFiles",
+    action="extend",
+    nargs="+",
+    metavar=["file1", "file2"],
+    help="One or multiple input files",
+)
+parser.add_argument(
+    "--compactFile", help="Compact detector file to use", type=str, default=""
+)
+
+reco_args = parser.parse_known_args()[0]
 
-from Configurables import MarlinProcessorWrapper, k4DataSvc, PodioInput, EDM4hep2LcioTool
 algList = []
+svcList = []
 
-evtsvc = k4DataSvc('EventDataSvc')
-evtsvc.input = ''
+evtsvc = k4DataSvc("EventDataSvc")
+svcList.append(evtsvc)
 
-inp = PodioInput('InputReader')
-inp.collections = [
-  'MCParticles',
-  'VertexBarrelCollection',
-  'VertexEndcapCollection',
-  'InnerTrackerBarrelCollection',
-  'OuterTrackerBarrelCollection',
-  'InnerTrackerEndcapCollection',
-  'OuterTrackerEndcapCollection',
-  'ECalEndcapCollection',
-  'ECalEndcapCollectionContributions',
-  'ECalBarrelCollection',
-  'ECalBarrelCollectionContributions',
-  'ECalPlugCollection',
-  'ECalPlugCollectionContributions',
-  'HCalBarrelCollection',
-  'HCalBarrelCollectionContributions',
-  'HCalEndcapCollection',
-  'HCalEndcapCollectionContributions',
-  'HCalRingCollection',
-  'HCalRingCollectionContributions',
-  'YokeBarrelCollection',
-  'YokeBarrelCollectionContributions',
-  'YokeEndcapCollection',
-  'YokeEndcapCollectionContributions',
-  'LumiCalCollection',
-  'LumiCalCollectionContributions',
-  'BeamCalCollection',
-  'BeamCalCollectionContributions',
-]
+if reco_args.compactFile:
+    compact_file = reco_args.compactFile
+else:
+    # add the same default as before
+    # TODO: maybe modernize this
+    compact_file = "CLICPerformance/Visualisation/CLIC_o3_v06_CED/CLIC_o3_v06_CED.xml"
 
-MyInitializeDD4hep = MarlinProcessorWrapper("MyInitializeDD4hep")
-MyInitializeDD4hep.OutputLevel = INFO
-MyInitializeDD4hep.ProcessorType = "InitializeDD4hep"
-MyInitializeDD4hep.Parameters = {
-                                 "DD4hepXMLFile": ["CLICPerformance/Visualisation/CLIC_o3_v06_CED/CLIC_o3_v06_CED.xml"]
-                                 }
+geoSvc = GeoSvc("GeoSvc")
+geoSvc.detectors = [compact_file]
+geoSvc.OutputLevel = INFO
+geoSvc.EnableGeant4Geo = False
+svcList.append(geoSvc)
 
-MyEventSelector = MarlinProcessorWrapper("MyEventSelector")
-MyEventSelector.OutputLevel = INFO
-MyEventSelector.ProcessorType = "EventSelector"
-MyEventSelector.Parameters = {
-                              "EventList": ["28", "0", "33", "0", "52", "0", "63", "0", "73", "0", "78", "0"]
-                              }
```

Comment:
This could be nice to have and should be fairly straight forward to do vi cl arg parsing? E.g.: https://github.com/AIDASoft/podio/blob/400f4f06bef3e7666c71e31be579ec715c412b8a/tools/podio-dump#L168-L175

Reply:
Getting this list as an argument yes. But I don't know how we would then select those?

We don't have an equivalent to the `<if>` construct in Marlin so we can not use the `EventSelector` processor. 
Is this possible with the `EvtSel` of the `ApplicationMgr`?
Or is there any other (already implemented) way?

Reply:
Ah right, I wasn't aware that the `EventSelector` uses the `if` machinery. Then my comment is moot and we will have to find another way to do this.